### PR TITLE
Add openai.chat connector

### PR DESCRIPTION
**Pull request by oshadajay**
# Description
Add openai.chat connector

Fixes: https://github.com/wso2-enterprise/choreo/issues/20370

### CODE REVIEW

File: openapi/openai.chat/Module.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
+## Overview
+
+This is a generated connector for the [OpenAI Chat API](https://beta.openai.com/docs/api-reference/chat) OpenAPI Specification. OpenAI is an American artificial intelligence research laboratory consisting of a non-profit corporation and a for-profit subsidiary. OpenAI conducts AI research with the declared intention of promoting and developing friendly AI. The OpenAI Chat API provides a way to access the state-of-the-art ChatGPT model (GPT-3.5) developed by OpenAI for a variety of tasks.
```
Comment (NipunaRanasinghe) [inline openapi/openai.chat/Module.md:4]: Do we need to mention that the connector supports only GPT-3.5 varients? Can't we specify other models (e.g GPT 4) in the completion request and retrieve responses in the same way?

File: openapi/openai.chat/Module.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        auth: {
+            token: "sk-XXXXXXXXX"
+        }
+    });
+```
+
+### Step 3: Invoke the connector operation
+1. Now you can use the operations available within the connector. 
+
+>**Note:** These operations are in the form of remote operations.
```
Comment (NipunaRanasinghe) [inline openapi/openai.chat/Module.md:37]:
```suggestion
>**Note:** These operations are in the form of Ballerina resource methods.
```

Comment (NipunaRanasinghe) [reply openapi/openai.chat/Module.md:37]: Or else we can remove the complete sentence to be consistent with the other module.mds.

Comment (NipunaRanasinghe) [reply openapi/openai.chat/Module.md:37]: Please check the other openai.* connector module.mds and do the same

File: openapi/openai.chat/Module.md (hunk 1 above)
Comment (oshadajay) [reply openapi/openai.chat/Module.md]: As of now, only GPT 3.5 Turbo model is supported for chat completions. Updated the content.

File: openapi/openai.chat/Module.md (hunk 2 above)
Comment (oshadajay) [reply openapi/openai.chat/Module.md]: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/775/commits/7b1a813be6589303915a604b00ff146808760f30.

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by aneeshafedo — APPROVED** [review]

**Review by abeykoon — APPROVED** [review]

**Pull request merged**